### PR TITLE
Change TCN model and recipe to fit TS forcasting task

### PR DESCRIPTION
**Pull request by TheaperDeng**
Related to issue #3300

### CODE REVIEW

File: pyzoo/zoo/automl/model/tcn.py [hunk 1]
```diff
@@ -88,33 +88,46 @@ def forward(self, x):
...
+                 input_feature_num,
+                 future_seq_len,
+                 output_feature_num,
+                 num_channels,
+                 kernel_size=3,
+                 dropout=0.1):
         super(TemporalConvNet, self).__init__()
+
+        if output_feature_num != num_channels[-1]:
+            num_channels[-1] = output_feature_num
```
Comment (yushan111) [inline pyzoo/zoo/automl/model/tcn.py:102]: append instead of changing value directly. Or add warning

File: pyzoo/zoo/automl/config/recipe.py [hunk 2]
```diff
@@ -100,6 +100,30 @@ def search_space(self, all_available_features):
...
+            "lr": 0.001,
+            "batch_size": 16,
+            "nhid": 8,
+            "levels": 8,
+            "kernel_size": 3,
+            "dropout": 0.1,
+            'past_seq_len': 80,
+            "input_feature_num": 2,
+            "future_seq_len": 40,
+            "output_feature_num": 2
```
Comment (yushan111) [inline pyzoo/zoo/automl/config/recipe.py:123]: We should only include hyper-parameters that need to be searched in search space.
We should ask user to pass these task configurations and we can also infer from input_x, input_y without user specified

File: pyzoo/zoo/automl/model/tcn.py [hunk 3]
```diff
@@ -88,33 +88,46 @@ def forward(self, x):
...
-            in_channels = input_size if i == 0 else num_channels[i - 1]
+            in_channels = input_feature_num if i == 0 else num_channels[i - 1]
             out_channels = num_channels[i]
             layers += [TemporalBlock(in_channels, out_channels, kernel_size,
                                      stride=1, dilation=dilation_size,
                                      padding=(kernel_size-1) * dilation_size, dropout=dropout)]
 
         self.tcn = nn.Sequential(*layers)
-        self.linear = nn.Linear(num_channels[-1], output_size)
+        self.linear = nn.Linear(past_seq_len, future_seq_len)
```
Comment (yushan111) [inline pyzoo/zoo/automl/model/tcn.py:115]: the input and output shape should be consistent with other models. Input of [num_samples, past_seq_len, feature_num], output of [num_samples, future_seq_len, target_num]

**Review by yushan111 — COMMENTED** [review]
Also, UT need to be added.

**Comment by TheaperDeng** [comment]
UT and review change is added. @yushan111

File: pyzoo/test/zoo/automl/model/test_tcn.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        y = np.random.randn(num_samples, output_time_steps, output_feature_dim)
+        return x, y
+
+    train_data = get_x_y(num_train_samples)
+    val_data = get_x_y(num_val_samples)
+    test_data = get_x_y(num_test_samples)
+    return train_data, val_data, test_data
+
+
+class TestVanillaLSTM(TestCase):
```
[the remark this answers is not in the document]
Comment (TheaperDeng) [reply pyzoo/test/zoo/automl/model/test_tcn.py:45]: hhhhh, sorry

**Comment by yushan111** [comment]
Note that automl pytorch model only support 3 dimension input and output for now.
Input: (num_samples, past_seq_len, input_feature_num)
output: (num_samples, future_seq_len, output_feature_num)

We will add 2-dimension output support in future PR

**Comment by TheaperDeng** [comment]
PR validation (combine) pass:
http://10.239.47.210:18888/job/ZOO-PR-Validation/4984/
http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests-py37-Spark3/395/

**Pull request merged**